js/custom_event.ts

```diff
@@ -0,0 +1,60 @@
+// Copyright 2018 the Deno authors. All rights reserved. MIT license.
+import * as domTypes from "./dom_types";
+import * as event from "./event";
+import { getPrivateValue } from "./util";
+
+// WeakMaps are recommended for private attributes (see MDN link below)
+// tslint:disable-next-line:max-line-length
+// https://developer.mozilla.org/en-US/docs/Archive/Add-ons/Add-on_SDK/Guides/Contributor_s_Guide/Private_Properties#Using_WeakMaps
+export const customEventAttributes = new WeakMap();
+
+export class CustomEventInit extends event.EventInit
+  implements domTypes.CustomEventInit {
+  detail = null;
+
+  constructor({
+    bubbles = false,
+    cancelable = false,
+    composed = false,
+    detail = null
```

Comment:
Actually, the problem is coming here.  The constructor signature looks like this because it is inferring the default arguments:

```ts
constructor(bubbles?: boolean, cancelable?: boolean, composed?: boolean, detail?: null)
```

This isn't what you want.  You should be explicit about the type of `detail` at least.